Source/Core/GroundPolylineGeometry.js

```diff
@@ -0,0 +1,1054 @@
+
+            // Encode start position and end position as high precision point + offset
+            var encodedStart = EncodedCartesian3.fromCartesian(startBottom, encodeScratch);
+            var forwardOffset = Cartesian3.subtract(endBottom, startBottom, offsetScratch);
+            var forward = Cartesian3.normalize(forwardOffset, rightScratch);
+
+            // Right plane
+            var startUp = Cartesian3.subtract(startTop, startBottom, startUpScratch);
+            startUp = Cartesian3.normalize(startUp, startUp);
+            var rightNormal = Cartesian3.cross(forward, startUp, rightScratch);
+            rightNormal = Cartesian3.normalize(rightNormal, rightNormal);
+
+            // Plane normals perpendicular to "geometry" normals, so cross (startTop - startBottom) with geometry normal at start
+            var startPlaneNormal = Cartesian3.cross(startUp, startGeometryNormal, startPlaneNormalScratch);
+            startPlaneNormal = Cartesian3.normalize(startPlaneNormal, startPlaneNormal);
+
+            // Similarly with (endTop - endBottom)
+            var endUp = Cartesian3.subtract(endTop, endBottom, endUpScratch);
+            endUp = Cartesian3.normalize(endUp, endUp);
+            var endPlaneNormal = Cartesian3.cross(endGeometryNormal, endUp, endPlaneNormalScratch);
+            endPlaneNormal = Cartesian3.normalize(endPlaneNormal, endPlaneNormal);
+
+            var texcoordNormalization3DX = segmentLength3D / length3D;
+            var texcoordNormalization3DY = lengthSoFar3D / length3D;
+
+            /** 2D **/
+            // In 2D case, positions and normals can be done as 2 components
+            var segmentLength2D = Cartesian3.distance(start2D, end2D);
+
+            var encodedStart2D = EncodedCartesian3.fromCartesian(start2D, encodeScratch2D);
+            var forwardOffset2D = Cartesian3.subtract(end2D, start2D, forwardOffset2DScratch);
+
+            // Right direction is just forward direction rotated by -90 degrees around Z
+            // Similarly with plane normals
+            var right2D = Cartesian3.normalize(forwardOffset2D, right2DScratch);
+            var swap = right2D.x;
+            right2D.x = right2D.y;
+            right2D.y = -swap;
+
+            var texcoordNormalization2DX = segmentLength2D / length2D;
+            var texcoordNormalization2DY = lengthSoFar2D / length2D;
+
+            /** Pack **/
+            for (j = 0; j < 8; j++) {
+                var vec4Index = vec4sWriteIndex + j * 4;
+                var vec2Index = vec2sWriteIndex + j * 2;
+                var wIndex = vec4Index + 3;
+
+                // Encode sidedness of vertex in texture coordinate normalization X
+                var sidedness = j < 4 ? 1.0 : -1.0;
+
+                // 3D
+                Cartesian3.pack(encodedStart.high, startHi_and_forwardOffsetX, vec4Index);
+                startHi_and_forwardOffsetX[wIndex] = forwardOffset.x;
```

Comment:
We usually don't put underscores in variable names, even local variables...except for constants in uppercase.